crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -50,11 +50,13 @@ def sync
   def activate
     return render_not_found if params[:platform].nil? || params[:repo].nil?
     respond_to do |format|
-      repo_object = Crowbar::Repository.where(platform: params[:platform], repo: params[:repo]).first
-      ret = ProvisionerService.new(logger).create_repository_item(repo_object) unless repo_object.nil?
-      if ret
+      ret = ProvisionerService.new(logger).enable_repository(params[:platform], params[:repo])
```

Comment:
If you are taking this strange behavior of Crowbar you should at least return a code and a message.